grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -500,35 +500,60 @@ class StockMovementService {
 
     void removeStockMovementItem(String id) {
         RequisitionItem requisitionItem = RequisitionItem.get(id)
-        removeRequisitionItem(requisitionItem)
+        ShipmentItem shipmentItem = ShipmentItem.get(id)
+        if (requisitionItem) {
+            removeRequisitionItem(requisitionItem)
+        } else {
+            removeShipmentItem(shipmentItem)
+        }
     }
 
     def getStockMovementItems(String id, String stepNumber, String max, String offset) {
         // FIXME should get stock movement instead of requisition
+        Shipment shipment = Shipment.get(id)
         Requisition requisition = Requisition.get(id)
         List<StockMovementItem> stockMovementItems = []
         List <RequisitionItem> requisitionItems = []
+        List <ShipmentItem> shipmentItems = []
```

Comment:
Do we need this outside the else scope? It feels like we're converting these to stockMovementItems before we return from that branch. And if that's the case for the requisitionItems, let's move that declaration into the if as well.